### DISCUSSION TITLE
edgetts

### DISCUSSION THREAD

**Discussion by giscus** [jianchang512/pyvideotrans#671] [Announcements]
# edgetts

免费开源使用简单的视频翻译配音软件-pyVideoTrans

https://pyvideotrans.com/edgetts

**Comment by LST-2003** [comment]
请问为什么部署完之后，edge-tts无法使用，报错信息如下：
获取edgeTTS角色失败Expecting value: line 1 column 1 (char 0)
Expecting value: line 1 column 1 (char 0)
而除配音外的其他功能均能正常使用

**Reply by jianchang512** [reply]
可能你的网络问题，无法连接到edge api

1. 下载使用最新版本软件
2. 重新打开如果仍有报错，请使用代理

**Comment by LST-2003** [comment]
实在不好意思，我今天在官网上重新下载源码部署之后还是出现同样的问题，因为我是英专生所以对于编程不太了解，想来请教您说的使用代理是什么意思呢？另外，我其实尝试写了一个简单的test文件，发现edgetts是可以把我预先输入的一段文件转成音频，但不知道在pyvideotrans里就会报错？




------------------&nbsp;原始邮件&nbsp;------------------
发件人: ***@***.***&gt;; 
发送时间: 2024年12月16日(星期一) 下午3:19
收件人: ***@***.***&gt;; 
抄送: ***@***.***&gt;; ***@***.***&gt;; 
主题: Re: [jianchang512/pyvideotrans] edgetts (Discussion #671)














 
可能你的网络问题，无法连接到edge api
  
下载使用最新版本软件
 
重新打开如果仍有报错，请使用代理

**Reply by jianchang512** [reply]
win系统还是mac系统
如果是win，下载那个3.5g的压缩包解压使用

源码部署执行时截图终端窗口输出信息

**Comment by WinsonPeng648** [comment]
你好，到了配音这一步报错，提示“可能被edge限流，请尝试使用或切换代理节点“。这个要怎么处理？我是科学上网可以登陆谷歌

**Reply by anrui8490028** [reply]
我也是同样的问题是什么原因啊

**Comment by fraise0324** [comment]
你好，到了配音这一步报错，提示“可能被edge限流，请尝试使用或切换代理节点“。这个要怎么处理？我是科学上网可以登陆谷歌

**Reply by anrui8490028** [reply]
我也是同样的问题

**Comment by Markzhan0754** [comment]
希望：为字幕配音——文本转语音，可以生成srt字幕文件。不然的话，每次文本配音后都需要音频再识别字幕，识别经常错误。感谢大佬制作

**Reply by yyf182** [reply]
支持加上这个功能

**Comment by jianchang512** [comment]
更新覆盖
https://github.com/jianchang512/pyvideotrans/releases/download/v3.59/win-PatchUpdate-3.61.7z

**Comment by chenpaner** [comment]
报错No module named 'videorans.edge_tts'用的是配音渠道是edge-TTS(免费),下载的版本号是3.64

**Reply by fy-kenny** [reply]
老版本没有我也是看到别人说的，我用的是3.56版本，如果你急着解决问题可以试试老版本

**Comment by FedericNi** [comment]
请问配音这部分，设否可以设定“，”和“。”的停顿时间，目前看“，”的时间合适，而“。”停顿的时间过长。导致文字转成语音后，到句号位置的停顿时间过长，听起来不连续。这个停顿时间可以设置吗？设置的位置在哪里，感谢！

**Comment by anrui8490028** [comment]
提示配音阶段出错：end time

**Comment by 15140946106** [comment]
之前可以正常使用edge-tts配音，中间一段时间未使用后，再次使用就提示404

**Reply by 15140946106** [reply]
404 Client Error: Not Found for url: https://edge.microsoft.com/translate/auth
Traceback (most recent call last):
  File "videotrans\translator\_base.py", line 68, in run
  File "videotrans\translator\_base.py", line 107, in _run_text
  File "tenacity\__init__.py", line 338, in wrapped_f
  File "tenacity\__init__.py", line 477, in __call__
  File "tenacity\__init__.py", line 378, in iter
  File "tenacity\__init__.py", line 421, in exc_check
tenacity.RetryError: RetryError[<Future at 0x1fe94413790 state=finished raised HTTPError>]

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "videotrans\task\only_one.py", line 83, in run
  File "videotrans\task\_stage_translate.py", line 31, in trans
  File "videotrans\translator\_runner.py", line 63, in run
  File "videotrans\translator\_base.py", line 74, in run
  File "tenacity\__init__.py", line 480, in __call__
  File "videotrans\translator\_microsoft.py", line 26, in _item_task
  File "requests\models.py", line 1026, in raise_for_status
requests.exceptions.HTTPError: 404 Client Error: Not Found for url: https://edge.microsoft.com/translate/auth
